internal/semver/semver.go

```diff
@@ -0,0 +1,49 @@
+package semver
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/hashicorp/go-version"
+	"github.com/hashicorp/vault/api"
+
+	"github.com/hashicorp/terraform-provider-vault/internal/consts"
+)
+
+func getTargetVaultVersion(ctx context.Context, client *api.Client) (string, error) {
+	resp, err := client.Sys().SealStatusWithContext(ctx)
+	if err != nil {
+		return "", err
+	}
+
+	if resp == nil {
+		return "", fmt.Errorf("expected response data, got nil response")
+	}
+
+	if resp.Version == "" {
+		return "", fmt.Errorf("key %q not found in response", consts.FieldVersion)
+	}
+
+	return resp.Version, nil
+}
+
+func GreaterThanOrEqual(ctx context.Context, client *api.Client, minVersionString string) (bool, string, error) {
```

Comment:
It might be nice to include a doc string for this function since it's exported, especially since it's not immediately apparent what all the return values are.

Reply:
Good call! Added a description for the function